src/main/java/net/minestom/server/entity/Entity.java

```diff
@@ -909,10 +909,16 @@ public void setVelocity(@NotNull Vec velocity) {
     /**
      * Gets if the entity currently has a velocity applied.
      *
-     * @return true if velocity is not set to 0
+     * @return Only false if the velocity is zero or the entity is on the ground and only moves downwards (gravity).
      */
     public boolean hasVelocity() {
-        return !velocity.isZero();
+        if (isOnGround()) {
+            // if the entity is on the ground and only "moves" downwards, it does not have a velocity.
+            return !(velocity.x() == 0 && velocity.z() == 0 && velocity.y() < 0);
```

Comment:
In my opinion, if an entity has a positive velocity, hasVelocity should return true immediatly and not just after one tick. 

The smaller check you suggested would make it inconsistent (People could set a velocity, check hasVelocity in the same method and would get false because the server didnt tick yet)

Reply:
should be `return velocity.x() != 0 || velocity.z() != 0 || velocity.y() > 0;`

currently when x, y and z of velocity are zero returns true